packages/client/src/utils/ShoppingCartContext/ShoppingCartContext.jsx

```diff
@@ -0,0 +1,64 @@
+import React, {
+  useState,
+  useContext,
+  useMemo,
+  useCallback,
+  createContext,
+} from 'react';
+import PropTypes from 'prop-types';
+import { useUserContext } from '../../userContext';
+
+const ShoppingCartContext = createContext({});
+
+export const useShoppingCart = () => {
+  return useContext(ShoppingCartContext);
```

Comment:
Here you should check if the context actually has a value because it might not have been wrapped properly.